app/Services/UserService.php

```diff
@@ -260,4 +266,32 @@ public function verifyEmail($data)
     return response()->json(['message' => 'Email verified successfully']);
 }
 
+public function enable2FA($userId)
+    {
+        $user = User::findOrFail($userId);
+        $user->update(['is_2fa_enabled' => true]);
+
+        return $this->generateOtp($user);
+    }
+    public function generateOtp($user)
+    {
+        $otp = Str::random(4);
+        $user->update(['otp' => $otp]);
```

Comment:
save the otp as encrypted in db also store the expire time
update([
            'otp' => bcrypt($otpGenerator),
            'otp_expires_at' => Carbon::now()->addMinutes((int) env('OTP_EXPIRE_MINS')),